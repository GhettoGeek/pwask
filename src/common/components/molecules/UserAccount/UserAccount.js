import React from 'react'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'
import { withTranslation } from 'react-i18next'
import {
  Link, Menu, MenuItem, Typography, IconButton,
} from '@material-ui/core'
import { AccountCircle as AccountIcon } from '@material-ui/icons'
import storage from '../../../utils/storage'

function UserAccount({ t }) {
  const [user, setUser] = React.useState({})
  const [isAuthenticated, setIsAuthenticated] = React.useState(false)
  const [anchorEl, setAnchorEl] = React.useState(null)

  async function getUserInfo() {
    const infos = await storage.get('auth')

    if (infos) {
      setUser({
        name: infos.userName,
        email: infos.userEmail,
      })
      setIsAuthenticated(true)
    }
  }

  function handleClick(event) {
    setAnchorEl(event.currentTarget)
  }

  function handleClose() {
    setAnchorEl(null)
  }

  React.useEffect(() => {
    getUserInfo()
  }, [user])

  return (
    <>
      <Typography color="secondary">
        {user.name}
      </Typography>
      <IconButton
        aria-owns={anchorEl ? 'simple-menu' : undefined}
        aria-haspopup="true"
        onClick={handleClick}
        color="secondary"
      >
        <AccountIcon fontSize="large" />
      </IconButton>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {
          isAuthenticated
            ? (
              ['dashboard', 'logout'].map((name) => (
                <MenuItem key={name}>
                  <Link component={RouterLink} to={`/${name}`}>
                    {t(`user.menu.${name}`)}
                  </Link>
                </MenuItem>
              ))
            )
            : (
              <MenuItem onClick={handleClose}>
                <Link component={RouterLink} to="/login">
                  {t('user.menu.login')}
                </Link>
              </MenuItem>
            )
        }
      </Menu>
    </>
  )
}

UserAccount.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(UserAccount)
