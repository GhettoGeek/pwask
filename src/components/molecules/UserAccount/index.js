import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'
import { withTranslation } from 'react-i18next'
import Link from '@material-ui/core/Link'
import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import Typography from '@material-ui/core/Typography'
import IconButton from '@material-ui/core/IconButton'
import AccountIcon from '@material-ui/icons/AccountCircle'
import storage from '../../../utils/storage'

class UserAccount extends Component {
  static propTypes = {
    t: PropTypes.func.isRequired,
  }

  state = {
    user: {},
    isAuthenticated: false,
    anchorEl: null,
  }

  componentDidMount() {
    this.getUserInfo()
  }

  getUserInfo = () => {
    const infos = storage.get('auth')

    if (infos) {
      this.setState({
        user: {
          name: infos.userName,
          email: infos.userEmail,
        },
        isAuthenticated: true,
      })
    }
  }

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget })
  }

  handleClose = () => {
    this.setState({ anchorEl: null })
  }

  render() {
    const { user, isAuthenticated, anchorEl } = this.state
    const { t } = this.props

    return (
      <>
        <Typography color="secondary">
          {user.name}
        </Typography>
        <IconButton
          aria-owns={anchorEl ? 'simple-menu' : undefined}
          aria-haspopup="true"
          onClick={this.handleClick}
          color="secondary"
        >
          <AccountIcon fontSize="large" />
        </IconButton>
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          {
            isAuthenticated
              ? (
                ['dashboard', 'logout'].map(name => (
                  <MenuItem key={name}>
                    <Link component={RouterLink} to={`/${name}`}>
                      {t(`user.menu.${name}`)}
                    </Link>
                  </MenuItem>
                ))
              )
              : (
                <MenuItem onClick={this.handleClose}>
                  <Link component={RouterLink} to="/login">
                    {t('user.menu.login')}
                  </Link>
                </MenuItem>
              )
          }
        </Menu>
      </>
    )
  }
}

export default withTranslation()(UserAccount)
