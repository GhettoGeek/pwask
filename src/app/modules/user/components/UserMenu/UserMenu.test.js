import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import UserAccount from './UserMenu'

const user = {
  name: 'User Test',
  email: 'user.test@mail.com',
}

describe('UserAccount', () => {
  let mockFunction

  beforeEach(() => {
    mockFunction = jest.fn()
    mockFunction.mockReturnValue('test')
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly the component', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <UserAccount user={user} onActions={mockFunction} />
      </MemoryRouter>,
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
