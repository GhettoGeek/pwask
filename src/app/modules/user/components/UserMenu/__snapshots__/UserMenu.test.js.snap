// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccount renders correctly the component 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-1 MuiIconButton-colorSecondary\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"icon-button\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root MuiIcon-fontSizeLarge\\" aria-hidden=\\"true\\" aria-haspopup=\\"true\\">account_circle</span></span></button>"`;
